src/controller/artifact/controller.go

```diff
@@ -229,11 +232,23 @@ func (c *controller) List(ctx context.Context, query *q.Query, option *Option) (
 		return nil, err
 	}
 
-	var artifacts []*Artifact
+	var artifacts []*artifact.Artifact
+	// Only the displayed accessory will in the artifact list
 	for _, art := range arts {
-		artifacts = append(artifacts, c.assembleArtifact(ctx, art, option))
+		accs, err := c.accessoryMgr.List(ctx, q.New(q.KeyWords{"ArtifactID": art.ID, "digest": art.Digest}))
+		if err != nil {
+			return nil, err
+		}
+		if len(accs) == 0 || (len(accs) > 0 && accs[0].Display()) {
```

Comment:
the Display() is hard code?

Reply:
base model under the accessory set it to false, as well as cosign.